### DISCUSSION TITLE
experiment setup

### DISCUSSION THREAD

**Discussion by aromajade** [pybamm-team/PyBaMM#4345] [Q&A]
can i setup experiment at different crate until different terminal voltage? is there any better way to setup it?
 for example the setup below. the simulation actually run the first charge till 3.8V, 2nd charge is skipped in the run for no reason. 
experiment = pybamm.Experiment(
    [
       "Charge at 0.5C until 3.8V",
       "Charge at 1C until 4.2V",
        "Rest for 1 hour",
    ]
)

**Comment by valentinsulzer** [comment]
If the event ("4.2V") is already triggered by the initial conditions, the step is skipped. For example, in your case, it could be that going from 0.5C at 3.8V to 1C makes the voltage go above 4.2V, in which case the step is skipped. To see if this is the case, you could try some lower C-rates or higher voltage limits in the second step.
Typically, when doing multi-step charging, the industry standard is to start with the highest C-rate then decrease, so I'm intrigued why you want to do the opposite (start low then increase)?

**Reply by aromajade** [reply]
yes, I agree. Thanks for helpful answers. I want to make sure whether pybamm support protocol like this by setting crate1 until a v1, then setting crate2 until a v2 without issues if my v1 is lower than v2. see example below, the 2nd charge at 0.2C is also skipped in the run. Do you have any ideas to setup this kind of protocol by controlling different c rates and target voltages.
i also tried: 
experiment = pybamm.Experiment(
[
"Charge at 1C until 3.8V",
"Charge at 0.2C until 4.2V",
"Rest for 1 hour",
]
)

**Reply by aromajade** [reply]
Problem solved by adding tuple. Thanks
